File: deposits-and-withdraw-private/withdrawal-addresses/withdrawal-addresses.js
require('dotenv').config();
const axios = require('axios');
const CryptoJS = require("crypto-js");

async function getWithdrawalAddress(asset, network) {
  const apiKey = process.env.API_KEY;
  const secretKey = process.env.API_SECRET;
  const timestamp = new Date().toISOString();
  const nonce = Math.random().toString(36).substring(2);
  const verb = 'GET';
  const path = 'api.ox.fun';
  const method = '/v3/withdrawal-addresses';
  const body = `asset=${asset}&network=${network}`;

  const msgString = `${timestamp}\n${nonce}\n${verb}\n${path}\n${method}\n${body}`;
  const sign = CryptoJS.enc.Base64.stringify(CryptoJS.HmacSHA256(msgString, secretKey));

  try {
    const response = await axios.get(`https://${path}${method}?${body}`, {
      headers: {
        'Content-Type': 'application/json',
        'AccessKey': apiKey,
        'Timestamp': timestamp,
        'Signature': sign,
        'Nonce': nonce
      }
    });

    if (response.data.success) {
      console.log('Withdrawal Address:', response.data.data);
    } else {
      console.error('Failed to fetch withdrawal address');
    }
  } catch (error) {
    console.error('Error making API request:', error.response ? error.response.data : error.message);
  }
}

// Example usage
getWithdrawalAddress('OX', 'Solana');

File: deposits-and-withdraw-private/withdrawal-addresses/withdrawal-addresses.py
import os
import datetime
import time
import base64
import hmac
import hashlib
import requests
from dotenv import load_dotenv

load_dotenv()

def get_withdrawal_address(asset, network):
    api_key = os.getenv('API_KEY')
    secret_key = os.getenv('API_SECRET').encode('utf-8')
    timestamp = datetime.datetime.utcnow().isoformat()
    nonce = str(int(time.time() * 1000))
    verb = 'GET'
    path = 'api.ox.fun'
    method = '/v3/withdrawal-addresses'
    body = f'asset={asset}&network={network}'

    msg_string = f'{timestamp}\n{nonce}\n{verb}\n{path}\n{method}\n{body}'
    sign = base64.b64encode(hmac.new(secret_key, msg_string.encode('utf-8'), hashlib.sha256).digest()).decode('utf-8')

    try:
        response = requests.get(f'https://{path}{method}?{body}', headers={
            'Content-Type': 'application/json',
            'AccessKey': api_key,
            'Timestamp': timestamp,
            'Signature': sign,
            'Nonce': nonce
        })
        response_data = response.json()
        if response_data.get('success'):
            print('Withdrawal Address:', response_data['data'])
        else:
            print('Failed to fetch withdrawal address')
    except requests.exceptions.RequestException as error:
        print('Error making API request:', error)

# Example usage
get_withdrawal_address('OX', 'Solana')
